Dockerfile

```diff
@@ -15,10 +14,17 @@ ENV PYTHONUNBUFFERED=1
 
 COPY --from=builder /app .
 
-RUN pip install --upgrade pip && \
-    pip install -r requirements.txt
+# Update and install dependencies with cleanup to reduce image size
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    libmariadb-dev \
+    && rm -rf /var/lib/apt/lists/* \
```

Comment:
It's good practice to clean up the apt cache after installing packages to reduce the image size. You are already doing this with `rm -rf /var/lib/apt/lists/*`, which is correct. However, ensure that `apt-get update` is run before installing packages, which you have done correctly.